slick-testkit/src/test/scala/slick/test/lifted/MySQLDMLTest.scala

```diff
@@ -17,11 +17,9 @@ class MySQLDMLTest {
       def value = column[String]("value")
       def * = (id, value) <> ((V.apply _).tupled, V.unapply)
     }
-    assertTrue("generates insert ignore statements",
-      TableQuery[T].insertOrUpdate(1).statements.mkString.startsWith("insert ignore"))
-    assertTrue("there are no update action if insert primary key only",
-      TableQuery[T].insertOrUpdate(1).statements.mkString.endsWith(")"))
+    assertTrue("generates query with update action if insert primary key only",
+      TableQuery[T].insertOrUpdate(1).statements.mkString.endsWith("on duplicate key update `id`=VALUES(`id`)"))
     assertTrue("generates query with update action if insert with value",
-      TableQuery[T2].insertOrUpdate(V("test", "test-value")).statements.mkString.endsWith("on duplicate key update `value`=VALUES(`value`)"))
+      TableQuery[T2].insertOrUpdate(V("test", "test-value")).statements.mkString.endsWith("on duplicate key update `id`=VALUES(`id`), `value`=VALUES(`value`)"))
```

Comment:
These tests don't actually hit the database. They only generate strings (that are supposed to be valid SQL) and check their contents in arbitrary way. There is infrastructure for real tests (i.e. ones that actually use database) in https://github.com/slick/slick/tree/main/slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests I think it would be worth it to add such a real test. Original issue had instructions on how to reproduce the problem.

Reply:
@tarsa
Implemented tests here. f8af6592ae6051af73fa1f2d20aaff1a305bd18f

Reply:
I kind of agree. Not saying to remove these tests at this point, but in general checking the strings sounds to me like testing the implementation rather than testing the behavior